### PR TITLE
feat: Update Default Gemini Base URL to v1beta

### PR DESCRIPTION
**Pull request by aliig**
In this PR, I updated the default base URL in the googleai_client.rs file to point to the beta version endpoint by changing "https://generativelanguage.googleapis.com/v1" to "https://generativelanguage.googleapis.com/v1beta". 

This mirrors the default implementation used in the [Google AI Python SDK](https://github.com/google-gemini/generative-ai-python/blob/8a29017e9120f0552ee3ad6092e8545d1aa6f803/google/generativeai/client.py#L60).

Functionally this opens up a larger variety of models including gemini 1.5 pro experimental from this month.

v1:
models/embedding-001
models/gemini-1.0-pro
models/gemini-1.0-pro-001
models/gemini-1.0-pro-latest
models/gemini-1.0-pro-vision-latest
models/gemini-1.5-flash
models/gemini-1.5-flash-001
models/gemini-1.5-flash-001-tuning
models/gemini-1.5-pro
models/gemini-1.5-pro-001
models/gemini-pro
models/gemini-pro-vision
models/text-embedding-004

v1beta:
models/aqa
models/chat-bison-001
models/embedding-001
models/embedding-gecko-001
models/gemini-1.0-pro
models/gemini-1.0-pro-001
models/gemini-1.0-pro-latest
models/gemini-1.0-pro-vision-latest
models/gemini-1.5-flash
models/gemini-1.5-flash-001
models/gemini-1.5-flash-001-tuning
models/gemini-1.5-flash-latest
models/gemini-1.5-pro
models/gemini-1.5-pro-001
models/gemini-1.5-pro-exp-0801
models/gemini-1.5-pro-latest
models/gemini-pro
models/gemini-pro-vision
models/text-bison-001
models/text-embedding-004

### CODE REVIEW

File: docs/docs/snippets/clients/providers/gemini.mdx [hunk 1]
```diff
@@ -88,4 +88,4 @@ client<llm> MyClient {
 </ParamField>
 
 
-For all other options, see the [official Google Gemini API documentation](https://ai.google.dev/api/rest/v1/models/generateContent).
+For all other options, see the [official Google Gemini API documentation](https://ai.google.dev/api/rest/v1beta/models/generateContent).
```
Comment (hellovai) [inline docs/docs/snippets/clients/providers/gemini.mdx:96]: is v1beta a superset to v1?

File: docs/docs/snippets/clients/providers/gemini.mdx [hunk 2]
```diff
@@ -4,7 +4,7 @@ slug: docs/snippets/clients/providers/gemini
 ---
 
 
-The `google-ai` provider supports the `https://generativelanguage.googleapis.com/v1/models/{model_id}/generateContent` and `https://generativelanguage.googleapis.com/v1/models/{model_id}/streamGenerateContent` endpoints.
+The `google-ai` provider supports the `https://generativelanguage.googleapis.com/v1beta/models/{model_id}/generateContent` and `https://generativelanguage.googleapis.com/v1beta/models/{model_id}/streamGenerateContent` endpoints.
```
Comment (hellovai) [inline docs/docs/snippets/clients/providers/gemini.mdx:7]: can you add a tip here that indicates why we are choosing v1beta instead of v1?

**Review by hellovai — CHANGES_REQUESTED** [review]
Hi @aliig this is a great PR, just some small nits

File: docs/docs/snippets/clients/providers/gemini.mdx (hunk 1 above)
Comment (aliig) [reply docs/docs/snippets/clients/providers/gemini.mdx:96]: Yes, at least it is currently! I haven't found that officially stated in my (limited) review of their docs but if you list the models in each v1beta has all the models in v1 and more.

Comment (hellovai) [reply docs/docs/snippets/clients/providers/gemini.mdx:96]: good enough for me. I think noting that the python library does this is a good point.

**Review by hellovai — APPROVED** [review]
LGTM!

**Comment by sxlijin** [comment]
+1, with the pace at which stuff is changing in the broader LLM landscape, Google's slowness in promoting v1beta to v1 is irrelevant for us here.

**Pull request merged**